lib/AST/DistributedDecl.cpp

```diff
@@ -271,12 +271,24 @@ swift::getDistributedSerializationRequirements(
   if (existentialRequirementTy->isAny())
     return true; // we're done here, any means there are no requirements
 
-  auto serialReqType = existentialRequirementTy->castTo<ExistentialType>()
-                           ->getConstraintType()
-                           ->getDesugaredType();
+  if (auto alias = dyn_cast<TypeAliasType>(existentialRequirementTy.getPointer())) {
+    auto ty = alias->getDesugaredType();
+    if (isa<ClassType>(ty) || isa<StructType>(ty) || isa<EnumType>(ty)) {
```

Comment:
There is still some unaddressed review feedback from the `main` PR related to this part of the change: https://github.com/apple/swift/pull/42348#discussion_r852211348

Can you please address the feedback?

Reply:
Sorry on the huge delay here and thanks for pointing out I missed that!

Back to compiler now post-WWDC and will fix here and on main 👍

Reply:
Thanks @hborla  -- I followed up here https://github.com/apple/swift/pull/59404 could you have a look and if it's correct I'll apply here as well 🙏

Reply:
Added the cleanup on main and here as well, thank you for noticing @hborla !